core/client/fs/src/main/java/alluxio/client/block/AlluxioBlockStore.java

```diff
@@ -145,39 +152,31 @@ public BlockInStream getInStream(long blockId, Protocol.OpenUfsBlockOptions open
           Preconditions.checkNotNull(options.getUfsReadLocationPolicy(),
               PreconditionMessage.UFS_READ_LOCATION_POLICY_UNSPECIFIED);
       address = blockLocationPolicy
-          .getWorker(GetWorkerOptions.defaults().setBlockWorkerInfos(getWorkerInfoList())
+          .getWorker(GetWorkerOptions.defaults().setBlockWorkerInfos(getAvailableWorkers())
```

Comment:
Will this interact poorly with some policies like `DeterministicHash`? Is it intended for `TieredLocality` a filter on top of the available workers before any location policy can make a choice?

Reply:
It should work properly. From the perspective of `DeterministicHashPolicy`, workers in different strict tiers don't exist. This is good because we never want cross-strict-tier traffic. That's the purpose of having strict tiers.